aiidalab/app.py

```diff
@@ -498,8 +505,8 @@ def install(
                     # Attempt rollback to previous version.
                     logger.info("Performing rollback to previously installed version.")
                     self._restore_from(trash_path)
-            except RuntimeError as error:
-                logger.warning(f"Rollback failed due to error: {error}!")
+            except RuntimeError as rerror:
```

Comment:
This is a rather confusing naming. Why was this change needed?

Reply:
Because there is another `error` lines ahead for another `RuntimeError`, the mypy complains this is not a good practice to override the local variable. I think using `type: igonre` here and replace it with `raise from` may be the correct solution.

Reply:
Please see the latest commit, I change the name to `inner_error`.
The issue is that the `error` outside is override by the inner `RuntimeError` exception and in the `finally` clause the it will not report the correct exception if the inner RuntimeError raised.

Reply:
@unkcpz thanks! I did not know about `raise ... from ...` syntax. I think that is a great solution in this case.
Go ahead with merging this.

I did notice one thing when looking at this code. The installation try block starting on line 465 is very big, and we're only catching the `RuntimeError` exceptions. Are we sure that there are no other failure modes coming from the functions that are called?
 
This is also most likely the direct reason for #334. When installing from command line, interrupting with Ctrl+C will interrupt the install without the rollback. I think until a more robust solution is implemented, as suggested by Sasha in #349, we should be catching `BaseException` instead of `RuntimeException`. Could you maybe take a look in another PR?

Reply:
Yes, those seem related, I'll give it a test and see if the exception caught here can fix the issue #334